### PR TITLE
AAFWriter: added support for AAF user comments (#22)

### LINKED ISSUE #22: AAFWriter: add support for comments
## Description

Currently the AAF writer doesn't translate user comments generated by the AAF reader (or manually by the user).

The adapter should translate comments from the clip / media reference onto the `MasterMob` and comments from the timeline onto the `CompositionMob`.

### PR DESCRIPTION
**Pull request by timlehr**
Fixes #22 

**Summarize your change.**

I added support for translating user comments in the OTIO metadata onto the mob objects.

- `clip["AAF"]["UserComments"]` will be written onto the clip `MasterMob`
- `clip.media_reference["AAF"]["UserComments"]` will be written onto the clip `MasterMob` and overwrites clip comments
-`timeline["AAF"]["UserComments"]` will be written onto the `CompositionMob`

**Reference associated tests.**

`test_aaf_adapter.py`

### CODE REVIEW

File: src/otio_aaf_adapter/adapters/aaf_adapter/aaf_writer.py [hunk 1]
```diff
@@ -140,6 +152,19 @@ def track_transcriber(self, otio_track):
...
+
+        user_comments = otio_item.metadata.get("AAF", {}).get("UserComments", {})
+        for key, val in user_comments.items():
+            if isinstance(val, int):
+                target_mob.comments[key] = val
+            elif isinstance(val, (float, Rational)):
+                target_mob.comments[key] = aaf2.rational.AAFRational(val)
+            else:
+                # ensure we can store comment value by converting it to unicode string
+                target_mob.comments[key] = str(val)
```
Comment (jminor) [inline src/otio_aaf_adapter/adapters/aaf_adapter/aaf_writer.py:166]: This else clause will catch strings, but also various object types, None, arrays, etc. Does using `str()` give you a Python description string for those? In particular it would be good to think through whether you want unexpected types to pass through as a best-guess string versus throwing an error. As a concrete example, I could imagine someone attempting to put a SerializableObject through here.

For strings, is there a length limit that should be checked?

**Review by jminor — CHANGES_REQUESTED** [review]

File: src/otio_aaf_adapter/adapters/aaf_adapter/aaf_writer.py (hunk 1 above)
Comment (timlehr) [reply src/otio_aaf_adapter/adapters/aaf_adapter/aaf_writer.py:166]: It should give us a string with varying quality - depending on how `__str__` is implemented for the object. I am not sure if there is a character limit for strings. @markreidvfx do you have the answer for that? Being able to put a SerializableObject object in there as JSON string would be handy to roundtrip studio-specific schemas.

Comment (markreidvfx) [reply src/otio_aaf_adapter/adapters/aaf_adapter/aaf_writer.py:166]: @timlehr I'll need to double check this, I don't think I've really tested property size limits in pyaaf. I naively assume its UINT16_MAX bytes but checking the the spec, it might actually be saying they want simple data properties to only be "64k bytes"?  [aaf-stored-format-spec](https://static.amwa.tv/ms-02-aaf-stored-format-spec.pdf). 

image

ugh it hurts my head every time I try and read this spec :p

This would be the size of the string in bytes when encoded to utf16le including 2 NULL terminator bytes.  
There is also a 17 byte header for indirect types like TaggedValues, so minus that too.

If the indirect type of the TaggedValue was set to a stream type, then the size limit would be around UINT64_MAX but I'm not sure if anything would be compatible with that other then pyaaf.

**Comment by timlehr** [comment]
@jminor @markreidvfx I changed the code to omit unsupported types of data in the comments with a warning. I also added some more test cases for unicode, lists, dicts and OTIO schemas.

**Review by markreidvfx — APPROVED** [review]
LGTM
